### DISCUSSION TITLE
Preload relationship not working

### DISCUSSION THREAD

**Discussion by lanthe2905** [adonisjs/core#3814] [Help]
hi everyone im newbiew annd my english to bad. To day i  use preload for relationship but can't get data. **Someone** can help me please

**_my code get:_** 

![image](https://user-images.githubusercontent.com/65931254/180791469-dfbdd240-3aa5-4a0b-9b4b-8bfb64c349e1.png)
**_result_**: 
![image](https://user-images.githubusercontent.com/65931254/180792185-eeb882d7-ed5b-45a5-bd4c-b8b11d18e179.png)

**_model_** 
![image](https://user-images.githubusercontent.com/65931254/180791666-9c3fd71b-85d2-40b8-a364-d50fee27565c.png)
![image](https://user-images.githubusercontent.com/65931254/180791797-edfe814c-eda1-4d12-8615-ea564b577d62.png)

**Comment by Barbapapazes** [comment]
Hello, 

Can you provide more information about this `for relationship but can't get data.` like a reproduction repo to view this issue ?

And have you populate (add) some data to the permissions table ?

**Reply by lanthe2905** [reply]
thank your answer, here is this my data (permissions) 
![image](https://user-images.githubusercontent.com/65931254/180810225-bdcd6c7f-e664-4483-9d9c-0d9bd443cc83.png)

**Reply by lanthe2905** [reply]
This is my migration
![image](https://user-images.githubusercontent.com/65931254/180810587-eedb7897-6945-4c5d-8211-18331390f94b.png)
![image](https://user-images.githubusercontent.com/65931254/180811793-f7c65aec-8c90-41c5-ab0d-deb3c9121427.png)

**Reply by WeyIin** [reply]
Maybe try
`table.bigInteger('group_permission_id').unsigned().references('id').inTable('group_permissions').onDelete('cascade')`

**Reply by thetutlage** [reply]
I see that your primary keys are integers and the foreign key is a big integer, therefore there is data type mis-match

**Reply by Barbapapazes** [reply]
And you can try to specify relation key explicitely ([view it on the doc](https://docs.adonisjs.com/reference/orm/relations/has-many#localkey))

**Reply by lanthe2905** [reply]
Thank Weylin i can do that

**Reply by lanthe2905** [reply]
> And you can try to specify relation key explicitely
hi  when i change type i can do that thank you so much
